.circleci/config.yml

```diff
@@ -0,0 +1,199 @@
+
+version: 2
+
+workflows:
+  version: 2
+  build-and-test:
+    jobs:
+      - build
+      - build-nightly
+      - build-lts-12
+      - build-lts-11
+      - build-lts-10
+      - build-lts-9
+
+
+references:
+
+  restore_backend_stack_cache: &restore_backend_stack_cache
+
+  # Default branch / master
+
+  default_cache_key: &default_cache_key
+    v1-stack-{{ checksum "stack.yaml" }}-{{ checksum "wai-middleware-throttle.cabal" }}
+
+  restore_stack_default_cache: &restore_stack_default_cache
+    restore_cache:
+      keys:
+        - *default_cache_key
+        - v1-stack-{{ checksum "stack.yaml" }}-
+        - v1-stack
+
+  save_stack_default_cache: &save_stack_default_cache
+    save_cache:
+      key: *default_cache_key
+      paths:
+        - ~/.stack
```

Comment:
Have you also considered `.stack-work`? This will cache the token bucket (and any other extra-deps).